src/data/real/ennreal.lean

```diff
@@ -1919,3 +1920,28 @@ by simpa only [image_image] using h.image_real_to_nnreal.image_coe_nnreal_ennrea
 
 end ord_connected
 end set
+
+namespace tactic
+
+open positivity
+
+private alias nnreal.coe_pos ↔ _ nnreal_coe_pos
+
+/-- Extension for the `positivity` tactic: cast from `ℝ≥0` to `ℝ`. -/
+@[positivity]
+meta def positivity_coe_nnreal : expr → tactic strictness
+| `(@coe _ _ %%inst %%a) := do
+  match inst with
```

Comment:
To protect against overlapping instances maybe the test should be for `tactic.is_def_eq` instead of pattern-matching? (Cf. Kazuhiko's paper on the new Coq ring tactic.)

Reply:
Can you elaborate? I'm not sure what using `is_defeq` would look like and how that lets me access `inst2` in
```
match inst with
| `(@coe_to_lift _ _ %%inst2)
end
```
There doesn't seem to be any mathlib use either, so maybe I should use `unify` instead?

Reply:
It would look something like (untested)!
```lean
do
  guard $ unify inst `(coe_to_lift _ _ (@coe_base _ _ nnreal.real.has_coe)),
  strictness_a ← core a,
  match strictness_a with
  | positive p := positive <$> mk_app ``nnreal_coe_pos [p]
  | nonnegative p := nonnegative <$> mk_app ``nnreal.coe_nonneg [a]
  | _, _ := failed
  end
```
You're right that `unify` is better here, because the expected shape of the expression has metavariables (and `is_def_eq` does not do meta assignments).

Reply:
Or does the expression quotation above need double backticks, triple backticks, or even explicit instantiation of its metavariables? I never know which quotation to use where.

Reply:
Not sure what the `guard` was supposed to do, but just removing it works.